File: Offer/Offer55.java
public class Offer55 {

    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public boolean isBalanced(TreeNode node){
        if (node == null){
            return true;
        }
        if(!isBalanced(node.left)){
            return false;
        }
        if (!isBalanced(node.right)){
            return false;
        }

        int left = getDepth(node.left);
        int right = getDepth(node.right);

        if(Math.abs(right-left) > 1){
            return false;
        }
        return true;
    }

        private int getDepth(TreeNode node){
        if (node == null){
            return 0;
        }
        int left = getDepth(node.left)+1;
        int right = getDepth(node.right)+1;

        return  right > left ? right:1;
    }
}

File: Offer/C++/offer55.cc
#include<iostream>
using namespace std;

struct TreeNode{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x):val(x),left(NULL),right(NULL){}
};

class Solution {

private:
bool ans = true;
public:
    bool isBalanced(TreeNode* root) {
        if(root == nullptr){
            return true;
        }
        int left = isBalanced(root->left);
        int right = isBalanced(root->right);

        if(abs(left-right)<=1){
            return true;
        }
        return false;
    }

    bool isBalanced(TreeNode* root){
        dfs(root);
        return ans;
    }

    int dfs(TreeNode* root){
        if(!root){
            return 0;
        }
        int left = 1+dfs(root->left);
        int right = 1+dfs(root->right);
        if(abs(left-right) > 1){
            ans = false;
        }
        return max(left,right);
    }
};